addons/sale_project/models/sale_order.py

```diff
@@ -76,26 +78,29 @@ def _search_tasks_ids(self, operator, value):
     def _compute_tasks_ids(self):
         tasks_per_so = self.env['project.task']._read_group(
             domain=['&', ('project_id', '!=', False), '|', ('sale_line_id', 'in', self.order_line.ids), ('sale_order_id', 'in', self.ids)],
-            groupby=['sale_order_id'],
+            groupby=['sale_order_id', 'state'],
             aggregates=['id:recordset', '__count']
         )
         so_with_tasks = self.env['sale.order']
-        for order, tasks_ids, tasks_count in tasks_per_so:
+        for order, state, tasks_ids, tasks_count in tasks_per_so:
             if order:
-                order.tasks_ids = tasks_ids
-                order.tasks_count = tasks_count
+                order.tasks_ids += tasks_ids
+                order.tasks_count += tasks_count
+                order.closed_task_count += state in ['1_done', '1_canceled'] and tasks_count
```

Comment:
```suggestion
                order.closed_task_count += state in CLOSED_STATES and tasks_count
```

to import like that: `from odoo.addons.project.models.project_task import CLOSED_STATES`

Reply:
:ok: